service/src/main/java/org/mskcc/smile/service/impl/SampleServiceImpl.java

```diff
@@ -66,6 +66,15 @@ public SmileSample saveSmileSample(SmileSample
                     sample.getSampleCategory().equals("research"))) {
                 LOG.info("Found updates to persist for sample: " + existingSample.getPrimarySampleAlias());
                 existingSample.updateSampleMetadata(sample.getLatestSampleMetadata());
+                // Updates investigatorId sampleAlias if sampleMetadata investigatorSampleId has been updated
+                if (existingSample.getDatasource().equals("igo")
+                        && !existingMetadata.getInvestigatorSampleId()
+                            .equals(sampleMetadata.getInvestigatorSampleId())) {
+                    existingSample.updateSampleAlias("investigatorId",
+                            sampleMetadata.getInvestigatorSampleId());
+                }
```

Comment:
In the even that a sample is being updated, I believe we would want to capture all updates to aliases as well. 

I'm thinking that we could add a method that instead will merge existing and new aliases.

The method should:
- know the existing aliases for the sample 
- should assume that the incoming sample metadata has updates to expected fields like: 
    - `investigatorSampleId` (instead of explicitly checking if there is an update to this field)
    - `igoId` (if research sample)
    - `dmpId` (if clinical sample)
- be agnostic of whether the sample is a clinical sample or research sample (meaning adjust the logic so that alias updates are applied to both clinical and research samples)

Reply:
This issue is also a good candidate for unit testing :)